pkg/agent/externalnode/external_node_controller.go

```diff
@@ -0,0 +1,777 @@
+		klog.ErrorS(err, "Failed to recover uplink name to the host interface name", "uplink", uplinkIfName, "hostInterface", hostIfName)
+		return err
+	}
+	// Move the IP configurations back to the host interface.
+	if err := util.MoveIFConfigurations(adapterConfig.IPs, adapterConfig.Routes, adapterConfig.MAC, adapterConfig.MTU, uplinkIfName, hostIfName); err != nil {
+		klog.ErrorS(err, "Failed to move back configuration to the host interface", "hostInterface", hostIfName)
+		return err
+	}
+	return nil
+}
+
+func getHostInterfaceName(iface v1alpha1.NetworkInterface) (string, []string, error) {
+	ifName := ""
+	ips := sets.NewString()
+	for _, ipStr := range iface.IPs {
+		var ipFilter *ip.DualStackIPs
+		ifIP := net.ParseIP(ipStr)
+		if ifIP.To4() != nil {
+			ipFilter = &ip.DualStackIPs{IPv4: ifIP}
+		} else {
+			ipFilter = &ip.DualStackIPs{IPv6: ifIP}
+		}
+		_, _, link, err := util.GetIPNetDeviceFromIP(ipFilter, sets.NewString())
+		if err == nil {
+			klog.InfoS("Using the interface", "linkName", link.Name, "IP", ipStr)
+			ips.Insert(ipStr)
+			if ifName == "" {
+				ifName = link.Name
+			} else if ifName != link.Name {
+				return "", ips.List(), fmt.Errorf("find different interfaces by IPs, ifName %s, linkName %s", ifName, link.Name)
+			}
+		}
+	}
+	if ifName == "" {
+		return "", ips.List(), fmt.Errorf("cannot find interface via IPs %v", iface.IPs)
+	}
+	return ifName, ips.List(), nil
+
+}
+
+func ParseHostInterfaceConfig(ovsBridgeClient ovsconfig.OVSBridgeClient, portData *ovsconfig.OVSPortData, portConfig *interfacestore.OVSPortConfig) (*interfacestore.InterfaceConfig, error) {
+	var interfaceConfig *interfacestore.InterfaceConfig
+	interfaceConfig = &interfacestore.InterfaceConfig{
+		InterfaceName: portData.Name,
+		Type:          interfacestore.ExternalEntityInterface,
+		OVSPortConfig: portConfig,
+	}
+	var hostUplinkConfig *interfacestore.EntityInterfaceConfig
+	if portData.ExternalIDs != nil {
+		entityIPStrs := strings.Split(portData.ExternalIDs[ovsExternalIDIP], eeSplitter)
+		var entityIPs []net.IP
+		for _, ipStr := range entityIPStrs {
+			entityIPs = append(entityIPs, net.ParseIP(ipStr))
+		}
+		interfaceConfig.IPs = entityIPs
+		uplinkName, _ := portData.ExternalIDs[ovsExternalIDUplinkName]
+		uplinkPortUUID, _ := portData.ExternalIDs[ovsExternalIDUplinkPort]
+		uplinkPortData, ovsErr := ovsBridgeClient.GetPortData(uplinkPortUUID, uplinkName)
+		if ovsErr != nil {
+			klog.Info("Failed to get uplink port data")
+			return nil, ovsErr
+		}
+		entityName, _ := portData.ExternalIDs[ovsExternalIDEntityName]
+		entityNamespace, _ := portData.ExternalIDs[ovsExternalIDEntityNamespace]
+		hostUplinkConfig = &interfacestore.EntityInterfaceConfig{
+			EntityName:      entityName,
+			EntityNamespace: entityNamespace,
+			UplinkPort: &interfacestore.OVSPortConfig{
+				PortUUID: uplinkPortUUID,
+				OFPort:   uplinkPortData.OFPort,
+			},
+		}
+	}
+	interfaceConfig.EntityInterfaceConfig = hostUplinkConfig
```

Comment:
This will be set to an empty value when there is no ExternalIDs set for the uplink port, with no error returned to the called.